frontend/components/FormComponents.tsx

```diff
@@ -0,0 +1,416 @@
+          )}
+        </div>
+        {newlyUploaded && (
+          <p className="is-size-7">(Submit this form to confirm upload)</p>
+        )}
+      </>
+    )
+  },
+)
+
+export const FormikAddressField = useFieldWrapper(
+  (props: BasicFormField & AnyHack): ReactElement => {
+    const { setFieldValue } = useFormikContext()
+    return (
+      <AddressField
+        addressValue={props.value}
+        changeAddress={(val) => setFieldValue(props.name, val)}
+      />
+    )
+  },
+)
+
+export const MultiselectField = useFieldWrapper(
+  ({
+    name,
+    choices,
+    placeholder,
+    value,
+    onBlur,
+    serialize,
+    deserialize,
+    valueDeserialize,
+    isMulti,
+  }: BasicFormField & AnyHack): ReactElement => {
+    const { setFieldValue } = useFormikContext()
+
+    const actualSerialize = (opt) => {
+      if (opt == null) {
+        return isMulti ? [] : null
+      }
+      if (serialize != null) {
+        return Array.isArray(opt) ? opt.map(serialize) : serialize(opt)
+      }
+      return opt.map(({ value, label }) => ({
+        id: value,
+        name: label,
+      }))
+    }
+
+    const actualDeserialize = (opt) => {
+      if (opt == null) {
+        return isMulti ? [] : null
+      }
+      if (deserialize != null) {
+        return Array.isArray(opt) ? opt.map(deserialize) : deserialize(opt)
+      }
+      return opt.map((item) => {
+        return {
+          value: item.id ?? item.value,
+          label: item.name ?? item.label,
+        }
+      })
+    }
+
+    return (
+      <Select
+        instanceId={name}
+        key={name}
+        placeholder={placeholder}
+        isMulti={isMulti}
+        value={(valueDeserialize ?? actualDeserialize ?? ((a) => a))(value)}
+        options={actualDeserialize(choices)}
+        onChange={(opt) => setFieldValue(name, actualSerialize(opt))}
+        onBlur={onBlur}
+        styles={{ container: (style) => ({ ...style, width: '100%' }) }}
+      />
+    )
+  },
+)
+
+export const CheckboxField = (
+  props: BasicFormField & AnyHack,
+): ReactElement => {
+  const { label, value, ...other } = props
+  const { status } = useFormikContext()
+  const errorMessage = status && status[props.name]
+  const fieldContext = useContext(FormFieldClassContext)
+  const isHorizontal = fieldContext.includes('is-horizontal')
+
+  const innerBody = (
+    <>
+      <div className="control">
+        <label className="checkbox">
+          <input type="checkbox" checked={value} {...other} /> {label}
+        </label>
+      </div>
+      {errorMessage && <p className="help is-danger">{errorMessage}</p>}
+    </>
+  )
+
+  return (
+    <div className={`field ${fieldContext}`}>
+      {isHorizontal ? (
+        <>
+          <div className="field-label">
+            <label className="label">{titleize(props.name)}</label>
+          </div>
+          <div className="field-body">{innerBody}</div>
+        </>
+      ) : (
+        innerBody
+      )}
+    </div>
```

Comment:
this file is awesome, in general tho some documentation on what each component does would be nice.